projects/evm-compatible-bridge/evm_contract_design.md

```diff
@@ -0,0 +1,141 @@
+  - For the chain native token, the token id will always be `0` so this call is unnecessary.
+- `crossTokens`
+  - This method will be called by the user to send tokens to the bridge.
+  - The bridge contract will check if the token is supported and if it is not, it will fail the transaction.
+  - The user will send tokens to the bridge contract address.
+  - The bridge will register this request with the federation contract.
+  - The bridge contract will emit an event to log the request.
+  - The amount of tokens must be valid according to the granularity rules.
+- `requestSupportForToken`
+  - The user will call this method to request support for a token (native to the EVM chain).
+  - This method will register a unique id and register this request with the federation contract.
+  - The bridge contract will emit an event to log the request.
+- All features supoprted by the ERC-1155 token standard.
+
+#### Admin methods
+
+This contract is "ownable" and the admin will be able to make some operations, for instance:
+
+- Contract suspension (i.e. pausable contract)
+  - In case of a security breach or legal obstruction we may have to halt operations.
+  - Example implementation [here](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/security/Pausable.sol)
+- Burn and mint tokens
+  - The admin should be able to manage tokens in the bridge.
+  - The admin can use this to refund tokens or make manual corrections.
+- Redeem fees
+  - The admin can redeem fees collected by the bridge from the transactions.
+- Change federation contract
+  - The admin can change the federation contract address.
+- Set token id
+  - The admin should be able to correct and change the mapping of equivalent tokens.
+  - This is to correct any mistakes or to manually support new tokens.
+
+#### Bridge contract storage
+
+##### _Token equivalency_
+
+Other than the storage required to implement the ERC-1155 token standard we require additional storage to track which tokens are allowed in the bridge and to map the equivalency of Hathor and EVM tokens.
+
+For tokens native to the EVM compatible chain we require an identification called `nativeTokenId` which we will map to a local id.
+This local id will be mapped to a Hathor token uid.
+The same will be done in reverse, a Hathor native token when crossing to the EVM chain will mint an equivalent token in our ERC-1155 contract.
+A token created in this contract will have a `uint256` id.
+
+```solidity
+struct nativeTokenId {
+  address contract;
+  uint256 tokenId; // Should be 0 for native tokens
+}
+// This nonce will begin at 2 and will increase with each new native token supported.
+// 0 is reserved for the native token of the chain.
+// 1 is reserved for the native token of Hathor.
+// Tokens created in the bridge contract will also increase this nonce.
+uint256 localIdNonce;
```

Comment:
This is used so we can implement an "auto increment" id.
This value will be initialized as 2 (since 0 and 1 are reserved) and any new token will use the value as his id then increase the value.

Reply:
I did not understand where this needs to be used. Do we need to make a mapping of ids like EVM <=> localIdNonce <=> Hathor id? Can't we make EVM <=> Hathor id directly?

Reply:
Closing this as we will only support fungible tokens and there's no need for it anymore.